### ISSUE TITLE
Unyson Front-End Builder

### ISSUE THREAD

**Issue by pear1** [ThemeFuse/Unyson#3237]
Hello,

I am still waiting on possibility to edit page content in front-end.
Are the any solutions to do that or is it planned to add this functionality? This feature would be very useful to all Unyson users.

**Comment by ViorelEremia** [comment]
At the moment we build a powerful frontend builder we hope this month will appear the first alpha version.

**Comment by pear1** [comment]
Great news! Can't wait for that to happen.
Please let me know if it can be tested somewhere.

**Comment by danyj** [comment]
Let us know if you need help with testing or anything

**Comment by danyj** [comment]
@ViorelEremia  all current shortcoe options would work in fpb?

**Comment by uxdesignit** [comment]
Great news! Can't wait for this ;)

**Comment by DimiBaitanciuc** [comment]
Great news guys. We have a lainch date for our visual builder called Brizy. 

We just launched the website yesterday si the actual launch of the builder later this month. 

Check out the website for a couple more details: https://brizy.io

**Comment by DimiBaitanciuc** [comment]
Regarding it working with the Unyson shortcodes from the back end builder that unyson currently has that is not possible. 

Brizy is built on a new way of thinking, a different approach on how you do things with it. 

There is a lot to talk about from the tech stand point of view (im not the one to answer those questions) but what I'll say is that we didn't want to build a DIVi here in the sense that we didn t want to extend the back end and star with certain constraints that will hurt the project in the long run.

The idea was to start with a clean slate and make the best builder we can without having the bagage of the old way of thinking. 

My personal opinion is that this will be the norm in building wordpress pages in a short while and it will be the default Unyson page builder in the months to come. 

I'm really excited with what we come up with and I"m sure you guys will love it yoo when we launch it later this month.

**Comment by pear1** [comment]
@DimiBaitanciuc Will it be somewhat possible to make older Unyson theme custom elements compatible with this new frontend builder?

**Comment by ViorelEremia** [comment]
@pear1 No, for the first versions is not possible maybe later we will make a mechanism that will allow you to expand the plugin's core.

**Comment by DimiBaitanciuc** [comment]
No, it will not be possible. We plan to create a developer API further down the line that will enable you to extend the builder to your liking. This has a high level of complexity on it's own since Brizy is fully built on React and we want to make it easy for you the developers to extend it. 

We've already taken into consideration the API and Brizy is built to support that in the future, but again that will happen further down the line. 

We'll keep you in the loop once we have something to share on this.

**Comment by danyj** [comment]
@DimiBaitanciuc  Nice for you but this 

>without having the bagage of the old way of thinking

is basically telling us that we all have wasted tremendous amount of time on something that you are trying to replace with Brizy.  Instead of combining Unsyon and Brizy  you are now creating competition between them.

**Comment by danyj** [comment]
@DimiBaitanciuc I dont doubt the quality and do not disagree with forward ideas , what I dont like now is that Brizy comes as independent plugin that puts a dark shadow on Unyson. This should have been a Unyson extension so that users should install Unsyon i order to have it. 

This way developers can opt out of the use like we have now for theme options , customizer or both and at the same time you increase the Use of  Unyson itself. 

Now we have the opposite effect where people will just install Brizy and would not like to use Unyson.

**Comment by DimiBaitanciuc** [comment]
I understand your point of view but I don t think Brizy will cast a shadow over Unyson. 

It s just a page builder after all. You'll still need the power of Unyson to create a full theme with header, footer, blogging etc. 

I know that is what we plan to do. Create themes on Unyson that has Brizy as the page builder. Plus of course continue to extend Brizy. 

Of course you can choose not to use Brizy. The current page builder will still be there. 
But why would you bother youself to put more work to figute out options for the builder. Shortcodes etc when you ll have them in brizy ready to be used with all the heavy lifting done by us. 

We ll still need to add some things to be up to par with other builders in terms of features (animations for ex or section dividers) but theyll be there eventually. Plus the api for developers sometime in the future. 

Maybe its only me but i see a lot of advantages here for all of us.

**Comment by sollerpolt** [comment]
Hi.

Just tested Brizy. It's really fine visual editor. Just have some questions:
1. Can I save my own block for reuse in another page? Just like user from GUI of your builder..
1.1. And like a developer I want to provide my own predefined blocks with theme.
2. Can I somehow define column width by numbers, not by D&D. It's hard to make 33% / 66%, for example.
3. Can I show element only on Desktop? I found a way to hide element on Desktop, but not to hide it on mobile.
4. Can I make global element? For example, If I have Block with partners logo and use it on multiple pages I want to have an ability to define it in one place and just reuse on another. Any changes will affect all pages with this element.
5. It will be great, if in font adding feature user can choose between predefined safe web fonts and Google Fonts. In lot of cases just Arial is suitable and are faster.
6. When and how can I add new elements and translations to your GREAT builder?
7. Is uploading images from WordPress native media gallery supported?

Tnx!

**Comment by DimiBaitanciuc** [comment]
Hi, 

Thanks for your kind words, we are very excited about the future. We think the vision of editing everything visually on the page is the right approach and this is the actual definition of visual editing. 

This term was so utilized in the past years in different forms, that it lost its meaning and it is somewhat diluted. We aim to bring that meaning of visual editing back with Brizy and I want to take this opportunity to thank you so much for testing it and asking all these questions because any suggestion and feedback will help us tremendously. 

Let me address your questions: 

1. Not right now but it is on the list and will be added after launch. 
1.1. we are working now to let you import export Brizy pages with your theme so your clients will be able to import Brizy pages made by you. I'll add an issue in our github regarding the developer predefined blocks, interesting idea, thanks. 

2. No you can't, I'll discuss it with the guys. We plan to create a way to add different empty column structures and most likely we'll have there a bunch of options to add different predefined widths (33% / 66%, 25% / 50% / 25%, and so on)

3. Yes you can show elements only on desktop. The Mobile editor can be accessed from here: http://jmp.sh/wdWzNsy and what you edit in this mobile instance will be applied only on mobile. 
On the options toolbar for every element there is an eye icon, that will toggle the display for that element: http://jmp.sh/3gCE8f1

4. The global elements are on the list as well. You can't do it right now but you will be able to in the future. 

5. Very interesting, I'll add this on the list as well. We usually tend to create opportunities for custom fonts and how to make things look fancy. This is the perfect example that we need to address the basic things as well.  

6. This is related to the API we plan to create to let developers add elements and functionalities. It's hard to say right now. We plan to have an API further down the road. This being a full React builder there are a set of complexities on how we'll tackle the API in such a way that will enable developers to extend the builder without necessarily having to know React in and out. We have some ideas here, more on this further down the road. 

7. Yes, it is supported. What you are testing on our website right now is only the React page builder, but the WordPress plugin (that was already accepted in the WP.org repository, not public yet) is communicating with the media library from WP and when you click to add an image the media library will pop up. 

Thanks for your feedback and please keep it coming.

**Comment by elbisnero** [comment]
The Brizi builder looks good, too bad it wasn't built to work with Unyson. I just had my theme approved on themeforest after a year of development, 20+ soft rejections, 6 months reviewing, and lots of hairloss: http://bit.ly/eventastic-theme I would really love to see a nice and simple Unyson front end builder. Ah, also Gutenberg compatibility as that seems to be where WordPress is headed...

**Comment by ViorelEremia** [comment]
@elbisnero about Gutenberg see here: https://github.com/ThemeFuse/Unyson/issues/3322

**Comment by elbisnero** [comment]
@ViorelEremia Thanks for the link :-)

**Comment by DimiBaitanciuc** [comment]
@elbisnero Thanks a bunch. Congrats on your theme acceptance. Envato has a very broken review system, they really need to take a good look at it and rethink it in my opinion. This is another discussion :) 

Regarding Brizy, I think you can switch all your demos pages on Brizy and bring the power of font end editing to your users. I'm sure they'll love it. This is what we'll do on ThemeFuse, we'll start migrating all our 53 themes on Brizy. We plan to keep what Unyson brings and re-create all the pages with Brizy.

We still need to add some small things like animations, better z-index control, custom sections, and a couple more but the majority are already in the works some already done and we test them in our dev environment.

By the way, the plugin is in beta now for anyone to take it for a spin: https://wordpress.org/plugins/brizy/

Known issues: 
-  issues on nginx servers (the icons don't load in the builder and there are some preview issues with the CSS) 
- some usability issues with the Publish button

Let us know if you have any feedback, suggestions. Thanks.

**Comment by elbisnero** [comment]
@DimiBaitanciuc Thanks for the heads up on the Brizy plugin. I will test it out and see how it applies to our current demos. If it works as expected I would gladly bundle it with our theme. Where can I get in touch with you to discuss further?

**Comment by DimiBaitanciuc** [comment]
@elbisnero Cool take it for a spin and see how it works. We can get in touch by email at dimi@themefuse.com.

**Comment by ThomasS4b** [comment]
I am extremly upset. Unyson is working in WP Liberty template and asked me today to install BRIZY. I agreed without research about BRIZY.

Now, my site is crashed. I don't have any longer my visual page builder elements BUT the complete HOME was converted to HTML and put in a text block. HOME is corrupted and there is only the menu and the footer.

How can I repair my site?

**Comment by ViorelEremia** [comment]
@ThomasS4b 
Send me some temporary credentials (admin, ftp): ```viorel@themefuse.com```
Please provide a link to this thread when submitting the message.

**Comment by FreshspaceHQ** [comment]
Hi, we're really disappointed. Effectively you've installed a plugin on all our Unyson sites without asking us, a bit like the U2 album release on iTunes!

The problem we've got is that we have clients who are editing content and seeing this nice bright pink Brizy button, they're pressing it and then it's loosing content from the page.

We're now deactivating and removing Brizy from all our sites to avoid a disaster.

**Comment by elbisnero** [comment]
@FreshspaceHQ I totally agree! its a bad move on their part, my customers are confused and are sending tickets asking if they need to use brizzy with our theme. What a disaster!

**Comment by hlaporthein** [comment]
@FreshspaceHQ @elbisnero . Yeah, Even I bought brizy pro, I actually have no reasons to add brizy builder in the website made by Unyson builder because One builder is enough for website. Why do we need to add in unyson plugin? I feel that Unyson is going with his own way and I know brizy is awesome, but, it shouldn't make unyson plugin confused for user experiences. I do accept Freemium software always come with advertisement but as a developer, I didn't know that brizy menu was in wordpress backend when it was launched, I got that updates when my customers told me what this "brizy" is.


Just sharing the things that I experienced.

**Comment by FreshspaceHQ** [comment]
Here's what happened when a client clicked on Brizy button and then tried to get back to Unyson editor...

![screen shot 2018-06-07 at 10 31 08](https://user-images.githubusercontent.com/36512002/41093571-96cd4866-6a43-11e8-8746-782010a39c0a.png)

**Comment by danyj** [comment]
@FreshspaceHQ @elbisnero @hlaporthein  guys , it was an oversight that was fixed with an update 2 days later. No ill intentions.  The current version has button only , not an actual install and you can hide it with CSS ad least.

**Comment by FreshspaceHQ** [comment]
That was done today with everything up to date.

**Comment by ThomasS4b** [comment]
Same here. The visual page builder was no longer available and deactivating Brizy did not bring it back. The only way to go back was to ring my provider and to import the backup from the day before.

**Comment by danyj** [comment]
@FreshspaceHQ , no ,  if your customer updated to  2.7.15 than the plugin was already installed and the button directs to it. Simply updating from 2.7.15 to 2.7.16 cant uninstall the Brizy plugin. 

 2.7.16  is showing the button only asking admin to try Brizy which when clicked installs the plugin

**Comment by i6tet** [comment]
Hi
I can not upload my images from the media library.
After selecting the image, only the preloader of the download is visible .. And nothing more happens, the image is not added.
What could be the problem?
